### PR TITLE
Add a simple one-entry-per-line file format for specifying symbol lists

### PR DESCRIPTION
**Pull request by sbc100**
For options such as `EXPORTED_FUNCTIONS` that take lists we now support
a simple file format that is just one line per symbol.   It doesn't
require any escaping or punctuation.

This is much easier to use and maintain and is in line with other compiler
toolchain options used by gcc and ld.  For example ld's
`--retain-symbols-file=filename` or wasm-ld's
`--undefined-file=filename`.

This should be especially useful for options like `ASYNCIFY_ADD` or
`ASYNCIFY_REMOVE` which often involve a lot of quoting (because they
include things like spaces and commas).

### CODE REVIEW

File: docs/emcc.txt [hunk 1]
```diff
@@ -123,16 +123,16 @@ Options that are modified or new in *emcc* are listed below:
...
+   a file. For example, the following will set "EXPORTED_FUNCTIONS"
+   based on the contents of the file at **path/to/file**.
 
       -s EXPORTED_FUNCTIONS=@/path/to/file
 
    Note:
 
-     * In this case the file might contain a JSON-formatted list of
-       functions: "["_func1", "func2"]".
+     * In this case the file should contain list of symbols, one per
```
Comment (kripken) [inline docs/emcc.txt:133]:
```suggestion
     * In this case the file should contain a list of symbols, one per
```

File: site/source/docs/tools_reference/emcc.rst [hunk 2]
```diff
@@ -112,15 +112,15 @@ Options that are modified or new in *emcc* are listed below:
...
+  You can also specify that the value of an option will be read from a file. For example, the following will set ``EXPORTED_FUNCTIONS`` based on the contents of the file at **path/to/file**.
 
   ::
 
     -s EXPORTED_FUNCTIONS=@/path/to/file
 
   .. note::
 
-    - In this case the file might contain a JSON-formatted list of functions: ``["_func1", "func2"]``.
+    - In this case the file should contain list of symbols, one per line.  For legacy use cases JSON-formatted files are also supported: e.g. ``["_func1", "func2"]``.
```
Comment (kripken) [inline site/source/docs/tools_reference/emcc.rst:123]:
```suggestion
    - In this case the file should contain a list of symbols, one per line.  For legacy use cases JSON-formatted files are also supported: e.g. ``["_func1", "func2"]``.
```

File: ChangeLog.md [hunk 3]
```diff
@@ -20,6 +20,10 @@ See docs/process.md for more on how version tagging works.
 
 2.0.21
 ------
+- Options such as EXPORTED_FUNCTIONS that can take a response file containing
+  list of symbols can now use a simple one-symbol-per-line format.  This new
+  format is much simpler and doesn't require commas between symbols, opening
+  or closing braces, or any kind of escaping for special characters.
```
Comment (kripken) [inline ChangeLog.md:26]:
```suggestion
  or closing braces, or any kind of escaping for special characters. The old
  JSON notation is still supported for backwards compatibility.
```

File: tests/test_other.py [hunk 4]
```diff
@@ -6226,12 +6226,16 @@ def test_dash_s_link_flag(self):
     self.assertContained('hello, world!', self.run_js('a.out.js'))
 
   def test_dash_s_response_file_string(self):
-    create_file('response_file', '"MyModule"\n')
-    self.run_process([EMXX, test_file('hello_world.cpp'), '-s', 'EXPORT_NAME=@response_file'])
+    create_file('response_file.txt', 'MyModule\n')
+    create_file('response_file.json', '"MyModule"\n')
```
Comment (kripken) [inline tests/test_other.py:6230]: is the newline needed in both cases? if so should we document that, as it seems like a pitfall? (but I hope it isn't needed)

Comment (sbc100) [reply tests/test_other.py:6230]: No, its not needed, but text files in general should always end in newlines.. (editors normally add one).. so including the newline is closer to what most uses in the wild will be like

**Review by kripken — APPROVED** [review]

**Comment by sbc100** [comment]
Did you add commits there?  (just a rebase?)

**Comment by kripken** [comment]
(yes, I happened to see you had automerge, and that it no longer merged cleanly due to a 1-line issue. so I fixed it for automerge's benefit, as otherwise I think you might not get a ping from github?)

**Comment by sbc100** [comment]
Sweet.. happy for you to fix to and land my PRs any time :)

**Pull request merged**